internal/registry/options.go

```diff
@@ -98,36 +97,23 @@ func (r *AuthOptionsGetter) GetOptions(ctx context.Context, repo *imagev1.ImageR
 			return nil, err
 		}
 		authenticator, authErr = secret.AuthFromSecret(authSecret, ref)
-	} else {
+	} else if provider != "generic" {
```

Comment:
```suggestion
	} else if provider := repo.GetProvider(); provider != "generic" {
```

Can declare the variable like this inside the else block since it's not used anywhere else

Reply:
Fixed via https://github.com/fluxcd/image-reflector-controller/pull/786/commits/a2b4049945aade03c5216671a8cf65d37ed5cdc6 .